applications/StructuralMechanicsApplication/custom_strategies/custom_schemes/explicit_multi_stage_kim_scheme.hpp

```diff
@@ -0,0 +1,989 @@
+     * @return Zero means  all ok
+     */
+    int Check(ModelPart& rModelPart) override
+    {
+        KRATOS_TRY;
+
+        BaseType::Check(rModelPart);
+
+        KRATOS_ERROR_IF(rModelPart.GetBufferSize() < 2) << "Insufficient buffer size for Central Difference Scheme. It has to be > 2" << std::endl;
+
+        KRATOS_ERROR_IF_NOT(rModelPart.GetProcessInfo().Has(DOMAIN_SIZE)) << "DOMAIN_SIZE not defined on ProcessInfo. Please define" << std::endl;
+
+        KRATOS_ERROR_IF((mDeltaTime.Fraction <= 0.0) || (mDeltaTime.Fraction >= 1.0)) << "fraction_delta_time must be >0 && <1 !" << std::endl;
+
+        return 0;
+
+        KRATOS_CATCH("");
+    }
+
+    /**
+     * @brief This is the place to initialize the Scheme. This is intended to be called just once when the strategy is initialized
+     * @param rModelPart The model of the problem to solve
+     */
+    void Initialize(ModelPart& rModelPart) override
+    {
+        KRATOS_TRY
+
+        ProcessInfo& r_current_process_info = rModelPart.GetProcessInfo();
+
+        // Preparing the time values for the first step (where time = initial_time +
+        // dt)
+        mTime.Delta = r_current_process_info[DELTA_TIME];
+        mTime.MidStep = mTime.Delta * mDeltaTime.Fraction;
+
+        /// Working in 2D/3D (the definition of DOMAIN_SIZE is check in the Check method)
+        const SizeType dim = r_current_process_info[DOMAIN_SIZE];
+
+        // Initialize scheme
+        if (!BaseType::SchemeIsInitialized())
+            InitializeExplicitScheme(rModelPart, dim);
+        else
+            SchemeCustomInitialization(rModelPart, dim);
+
+        BaseType::SetSchemeIsInitialized();
+
+        KRATOS_CATCH("")
+    }
+
+    /**
+     * @brief It initializes time step solution. Only for reasons if the time step solution is restarted
+     * @param rModelPart The model of the problem to solve
+     * @param rA LHS matrix
+     * @param rDx Incremental update of primary variables
+     * @param rb RHS Vector
+     * @todo I cannot find the formula for the higher orders with variable time step. I tried to deduce by myself but the result was very unstable
+     */
+    void InitializeSolutionStep(
+        ModelPart& rModelPart,
+        TSystemMatrixType& rA,
+        TSystemVectorType& rDx,
+        TSystemVectorType& rb
+        ) override
+    {
+        KRATOS_TRY
+        BaseType::InitializeSolutionStep(rModelPart, rA, rDx, rb);
+        InitializeResidual(rModelPart);
+        KRATOS_CATCH("")
+    }
+
+    /**
+     * @brief It initializes the non-linear iteration
+     * @param rModelPart The model of the problem to solve
+     * @param rA LHS matrix
+     * @param rDx Incremental update of primary variables
+     * @param rb RHS Vector
+     * @todo I cannot find the formula for the higher orders with variable time step. I tried to deduce by myself but the result was very unstable
+     */
+    void InitializeNonLinIteration(
```

Comment:
is this different from the version in the core?

Reply:
no, there is a PR pending : #5121